### PR TITLE
add mars common API

### PR DESCRIPTION
**Pull request by NJUPole**
Add common API, it will be used by all user's interaction with cluster.

For example, web server hold a `MarsAPI` and all the handler get cluster information via it.

resolve #26

### CODE REVIEW

File: mars/api.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    def get_schedulers_info(self):
+        schedulers = self.cluster_info.get_schedulers()
+        infos = []
+        for scheduler in schedulers:
+            info_ref = actor_client.actor_ref(NodeInfoActor.default_name(),
+                                              address=scheduler)
+            infos.append(info_ref.get_info())
+        return infos
+
+    def list_workers(self):
```
Comment (wjsi) [inline mars/api.py]: list_xxx should not return a number

File: mars/api.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+from .scheduler.session import SessionManagerActor
+from .scheduler.graph import ResultReceiverActor
+
+logger = logging.getLogger(__name__)
+
+actor_client = new_client()
+
+
+class MarsAPI(object):
+    def __init__(self, scheduler_ip):
```
Comment (chineking) [inline mars/api.py]: actor_client can be instantiated each time in the MarsAPI init.

File: mars/api.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    def submit_graph(self, session_id, serialized_graph, graph_key, target):
+        session_uid = SessionActor.gen_name(session_id)
+        session_ref = self.get_actor_ref(session_uid)
+        session_ref.submit_tensor_graph(serialized_graph, graph_key, target, _tell=True)
+
+    def delete_graph(self, session_id, graph_key):
+        graph_uid = GraphActor.gen_name(session_id, graph_key)
+        graph_ref = self.get_actor_ref(graph_uid)
+        graph_ref.stop_graph()
```
Comment (chineking) [inline mars/api.py]: stop graph in the delete seems unreasonable, how about add a stop_graph and do destroy in the delete_graph?

File: mars/session.py [hunk 4]
```diff
@@ -43,10 +43,9 @@ class Session(object):
 
     def __init__(self, endpoint=None):
         if endpoint is not None:
-            from .web import get_client
+            from .web.session import Session
```
Comment (chineking) [inline mars/session.py]: Here must be 3 kinds of sessions, besides the LocalSession, if the endpoint is a http/https url, the web session required, but for the LocalCluster, web is optional, if the endpoint is the scheduler address, the session should be based on the actor client api.

Comment (chineking) [reply mars/session.py]: A LocalClusterSession can be added to do the job.

**Pull request merged**